test/chartist.component.spec.ts

```diff
@@ -156,24 +149,22 @@ describe('chartist component', function(): void {
     });
   });
 
-  it('should throw an error when missing type', function(): void {
+  it('should throw an error when missing type', function (): void {
     instance.data = data['Bar'];
 
     expect(instance.ngOnInit).toThrow();
   });
 
-  it('should throw an error when missing data', function(): void {
+  it('should throw an error when missing data', function (): void {
     instance.type = 'Bar';
 
     expect(instance.ngOnInit).toThrow();
   });
 
-  it('should throw an error when an invalid chart type is passed', function() {
+  it('should throw an error when an invalid chart type is passed', function (): void {
     instance.data = data['Bar'];
-    instance.type = 'NotAChart';
-
     instance.renderChart().catch((err) => {
-      expect(err.message).toBe('NotAChart is not a valid chart type');
+      expect(err.message).toBe('undefined is not a valid chart type');
```

Comment:
I guess the change from `var` to `let` is fine, even if it is unrelated, so you can leave that.

But yeah, I would like everything else reverted please.

As far as the "erroneous" tests goes:

> This "fixed" test was also due to a ts error (TS2322):

That is quite intentional, I understand that TS will throw an error, as it should. But the test is designed to test the code path, not the TS compiler since this all gets transpiled into JS at the end of the day. Personally, I'd prefer if you revert that change.